src/three-idx.ts

```diff
@@ -74,40 +85,52 @@ export class ThreeIDX {
     return this._v03ID || `did:3:${this.docs.threeId.id.baseID.toString()}`
   }
 
-  async create3idDoc(docParams: Record<string, any>): Promise<void> {
-    this.docs.threeId = await this.ceramic.createDocument('tile', docParams, {
-      anchor: false,
-      publish: false,
-    })
+  async create3idDoc(docParams: ThreeIdState): Promise<void> {
+    this.docs.threeId = await TileDocument.create(
+      this.ceramic,
+      docParams.content,
+      docParams.metadata,
+      {
+        anchor: false,
+        publish: false,
+      }
+    )
   }
 
   get3idVersion(): string {
-    const docId = this.docs.threeId.anchorCommitIds.pop()
+    const anchorCommitIds = this.docs.threeId.anchorCommitIds
+    const docId = anchorCommitIds[anchorCommitIds.length - 1]
```

Comment:
I presume there might have been a bug here by using `.pop()` before? Nice catch if so.

Reply:
Not a bug, it modifies the array, which smells bad.